sku.0/sys.server/compiled/game/script/gm/cmd.java

```diff
@@ -654,86 +654,149 @@ else if (faction.equalsIgnoreCase("rebel"))
     }
     public int cmdGetRank(obj_id self, obj_id target, String params, float defaultTime) throws InterruptedException
     {
-        if (!isIdValid(target) || !isPlayer(target))
+		if (!isIdValid(target) || !isPlayer(target))
         {
+            sendSystemMessageTestingOnly(self, "Syntax: /getRank -target");
+            sendSystemMessageTestingOnly(self, "Because you did not specify a target, you were used.");
             target = self;
         }
         int rank = pvpGetCurrentGcwRank(target);
-        String faction = factions.getFaction(target);
-        sendSystemMessageTestingOnly(self, "(" + target + ") " + getName(target) + "'s rank = (" + rank + ") " + getString(new string_id("faction_recruiter", factions.getRankName(rank, faction))));
+        int faction = pvpGetAlignedFaction(target);
+        String factionName = null;
+		
+		if (faction == -615855020) {
+			factionName = "Imperial";
+		} else if (faction == 370444368) {
+			factionName = "Rebel";
+		}
+		
+		if (faction != 0) {
+            sendSystemMessageTestingOnly(self, "The rank of "+getPlayerName(target)+" ("+target+") is "+rank+" ("+factionName+")");
+        } else {
+            sendSystemMessageTestingOnly(self, getPlayerName(target)+ " is not aligned with a faction.");
+        }
         return SCRIPT_CONTINUE;
+		
     }
     public int cmdSetRank(obj_id self, obj_id target, String params, float defaultTime) throws InterruptedException
     {
-        if (utils.hasScriptVar(self, "setRank.pid"))
+
+		int originalRank = pvpGetCurrentGcwRank(target);
+        int currentRank = pvpGetCurrentGcwRank(target);
+        int faction = pvpGetAlignedFaction(target);
+        String command = null;
+        String rankName = null;
+
+        if (params.contains(gm.KEYWORD_TARGET))
         {
-            int oldpid = utils.getIntScriptVar(self, "setRank.pid");
-            sui.closeSUI(self, oldpid);
-            utils.removeScriptVarTree(self, "setRank");
+            params = gm.removeKeyword(params, gm.KEYWORD_TARGET);
         }
-        dictionary d = gm.parseTarget(params, target, "SETRANK");
-        if (d == null)
+        if (params == null || params.equalsIgnoreCase(""))

```

Comment:
`params.equalsIgnoreCase` is how all of the other commands in this script are written (I just copied and pasted it). I believe `isEmpty` doesn't work because of the tokenizer.

Reply:
A lot of the code written by SOE couldn't take advantage of newer methods like the one Ios is recommending.  So he's right, isEmpty() is the better implementation besides what was written in the 2000's.

Reply:
> `params.equalsIgnoreCase` is how all of the other commands in this script are written (I just copied and pasted it). I believe `isEmpty` doesn't work because of the tokenizer.

Yeah, unfortunately it is everywhere, and they should probably be replaced in the future. Tokenizer is called on the string, for which `isEmpty()` is a method in the String class, so it won't cause any issues.